File: app/src/main/java/ai/elimu/analytics/dao/StoryBookLearningEventDao.java
package ai.elimu.analytics.dao;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

import ai.elimu.analytics.entity.StoryBookLearningEvent;

@Dao
public interface StoryBookLearningEventDao {

    @Insert
    void insert(StoryBookLearningEvent storyBookLearningEvent);

    @Query("SELECT * FROM StoryBookLearningEvent")
    List<StoryBookLearningEvent> loadAll();
}

File: app/src/main/java/ai/elimu/analytics/dao/StoryBookLearningEventDao.kt
package ai.elimu.analytics.dao

import ai.elimu.analytics.entity.StoryBookLearningEvent
import androidx.room.Dao
import androidx.room.Insert
import androidx.room.Query

@Dao
interface StoryBookLearningEventDao {
    @Insert
    fun insert(storyBookLearningEvent: StoryBookLearningEvent)

    @Query("SELECT * FROM StoryBookLearningEvent ORDER BY timestamp ASC")
    fun loadAllOrderedByTimestampAsc(): List<StoryBookLearningEvent>

    @Query("SELECT COUNT(*) FROM StoryBookLearningEvent")
    fun getCount(): Int
}
